circe/src/main/scala/org/http4s/circe/CirceEntityDecoder.scala

```diff
@@ -0,0 +1,14 @@
+package org.http4s.circe
+
+import cats.effect.Sync
+import io.circe.Decoder
+import org.http4s.EntityDecoder
+
+/**
+  * Derive [[EntityDecoder]] if implicit [[Decoder]] is in the scope without need to explicitly call `jsonOf`
+  */
+trait CirceEntityDecoder {
```

Comment:
Why do this as a trait and then an object which extends said trait? Why not just place this in the object?